### PR TITLE
Add load-time configuration to resize the node creation buffer

### PR DESCRIPTION
**Pull request by jeffreylovitz**
This PR adds the load-time configuration `NODE_CREATION_BUFFER` to allow users to modify the amount of empty space is reserved in matrices for future node creations.

### CODE REVIEW

File: tests/unit/test_graph.cpp [hunk 1]
```diff
@@ -57,6 +57,7 @@ class GraphTest : public ::testing::Test {
 		GrB_Index ncols, nrows, nvals;
 
 		// Create nodes.
+		Graph_AllocateNodes(g, node_count);
```
Comment (swilly22) [inline tests/unit/test_graph.cpp:60]: Why was this call added?

**Review by swilly22 — CHANGES_REQUESTED** [review]
Let's keep this PR simple
Today our matrices dimensions (NxN) are equal to the graph's node datablock itemCap, which is: number of blocks * DATABLOCK_BLOCK_CAP
I believe the idea was to allow the user to specify a different value which will become a replacement to the `DATABLOCK_BLOCK_CAP` specifically for the graph node's datablock.

consider
```c
DataBlock *DataBlock_New
(
    uint64_t blockCap,    // max number of items in a single block
    uint64_t itemCount,   // initial number of items required
    uint itemSize, fpDestructor fp
);
```

to be compact we can use a small value for the `blockCap` in addition to passing `0` as `itemCount`
```c
DataBlock *DataBlock_New(128, 0, (fpDestructor)FreeEntity);
```

**Review by swilly22 — APPROVED** [review]

File: src/util/datablock/datablock.c [hunk 2]
```diff
@@ -40,45 +45,63 @@ static void _DataBlock_AddBlocks(DataBlock *dataBlock, uint blockCount) {
...
-	dataBlock->itemSize = itemSize + ITEM_HEADER_SIZE;
-	dataBlock->blockCount = 0;
-	dataBlock->blocks = NULL;
-	dataBlock->deletedIdx = array_new(uint64_t, 128);
-	dataBlock->destructor = fp;
+	dataBlock->blocks      =  NULL;
+	dataBlock->itemSize    =  itemSize + ITEM_HEADER_SIZE;
+	dataBlock->itemCount   =  0;
+	dataBlock->blockCount  =  0;
+	dataBlock->blockCap    =  blockCap;
```
Comment (AviAvni) [inline src/util/datablock/datablock.c:94]: nit: ,ove 2 lines up

File: src/util/datablock/datablock.c [hunk 3]
```diff
@@ -13,23 +13,28 @@
 #include <stdbool.h>
 
 // Computes the number of blocks required to accommodate n items.
-#define ITEM_COUNT_TO_BLOCK_COUNT(n) \
-    ceil((double)n / DATABLOCK_BLOCK_CAP)
+#define ITEM_COUNT_TO_BLOCK_COUNT(n, cap) \
```
Comment (AviAvni) [inline src/util/datablock/datablock.c:16]: why need cap? can't we just write dataBlock->blockCap?

File: src/util/datablock/oo_datablock.h [hunk 4]
```diff
@@ -27,3 +27,4 @@ void *DataBlock_AllocateItemOutOfOrder(DataBlock *dataBlock, uint64_t idx);
  * 1. Item distructor is not invoked in this call.
  * 2. This call does not decrease the number of items in the data block. */
 void DataBlock_MarkAsDeletedOutOfOrder(DataBlock *dataBlock, uint64_t idx);
```
Comment (AviAvni) [inline src/util/datablock/oo_datablock.h:29]: if the c file deleted why not delete this file too?

Comment (swilly22) [reply src/util/datablock/oo_datablock.h:29]: clear separation of the API, but we could delete this one as-well, I just wanted to keep this API separated.

File: src/util/datablock/datablock.c (hunk 3 above)
Comment (swilly22) [reply src/util/datablock/datablock.c:16]: this macro answers the question:
given `n` items how many blocks do we need

**Pull request merged**